src/neo/Wallets/Wallet.cs

```diff
@@ -527,22 +527,33 @@ public long CalculateNetworkFee(DataCache snapshot, Transaction tx)
             int size = Transaction.HeaderSize + tx.Signers.GetVarSize() + tx.Attributes.GetVarSize() + tx.Script.GetVarSize() + IO.Helper.GetVarSize(hashes.Length);
             uint exec_fee_factor = NativeContract.Policy.GetExecFeeFactor(snapshot);
             long networkFee = 0;
+            int index = -1;
             foreach (UInt160 hash in hashes)
             {
+                index++;
                 byte[] witness_script = GetAccount(hash)?.Contract?.Script;
+                byte[] invocationScript = null;
 
-                if (witness_script is null && tx.Witnesses != null)
+                if (tx.Witnesses != null)
                 {
-                    // Try to find the script in the witnesses
-
-                    foreach (var witness in tx.Witnesses)
+                    if (witness_script is null)
                     {
-                        if (witness.ScriptHash == hash)
+                        // Try to find the script in the witnesses
+
+                        foreach (var witness in tx.Witnesses)
                         {
-                            witness_script = witness.VerificationScript;
-                            break;
+                            if (witness.ScriptHash == hash)
+                            {
+                                witness_script = witness.VerificationScript;
+                                invocationScript = witness.InvocationScript;
+                                break;
+                            }
```

Comment:
```suggestion
                            if (witness.ScriptHash == hash)
                            {
                                witness_script = witness.VerificationScript;
                                break;
                            }
```
I.e. if witness.ScriptHash == hash then it's not a contract witness and we **don't need to know invocation script**, because we know exactly its length in case of signature or multisignature witness.

Reply:
The execution cost of the verification could be different with different invocation scripts

Reply:
Yes, but that's bothering us only for contract-based witnesses. But when witness scripthash is the same as the signer's hash, then it's either standard signature or standard multisignature witness (not a contract-based witness). In both cases, we know the exact cost of invocation/verification because we know how sig/multisig invocation/verification scripts looks like:
```
else if (witness_script.IsSignatureContract())
{
    size += 67 + witness_script.GetVarSize();
    networkFee += exec_fee_factor * SignatureContractCost();
}
else if (witness_script.IsMultiSigContract(out int m, out int n))
{
    int size_inv = 66 * m;
    size += IO.Helper.GetVarSize(size_inv) + size_inv + witness_script.GetVarSize();
    networkFee += exec_fee_factor * MultiSignatureContractCost(m, n);
}
```
So you see that we don't use the actual invocation script in the code above in case of sig/multisig witness. Consequently, we don't need invocation script to be set when witness scripthash equals to the signer's hash:
```suggestion
                            if (witness.ScriptHash == hash)
                            {
                                witness_script = witness.VerificationScript;
                                break;
                            }
```

Reply:
> But when witness scripthash is the same as the signer's hash, then it's either standard signature or standard multisignature witness (not a contract-based witness).

That is true.

Reply:
If it's standard contract the invocationScript it won't take the invocationScript, it's in a different condition